src/Modal/users/UserModal.js

```diff
@@ -0,0 +1,138 @@
+import React, { Fragment } from "react";
+import Modal from "react-bootstrap/Modal";
+import { updateUserRequest, createNewUser } from "../../services/api";
+import { getUserObject } from "../../utils/helpers/generatorHelper";
+
+function EditUser({
+  user,
+  setUser,
+  showModal,
+  setShowModal,
+  users,
+  setUsers,
+  title,
+  type,
+}) {
+  const addUserHandler = async () => {
+    if (user.name) {
+      setShowModal(false);
+      const payLoad = getUserObject(user);
+      const response = await createNewUser(payLoad);
+      const { data } = response;
+      setUsers([...users, data]);
+    }
+  };
+  const payLoad = user;
+  const usersCopy = [...users];
+  const updateUser = async () => {
+    setShowModal(false);
+    try {
+      const userIndex = usersCopy.findIndex((user) => user.id === payLoad.id);
+      const userNewData = getUserObject(payLoad);
+      usersCopy[userIndex] = userNewData;
+      setUsers([...usersCopy]);
+      await updateUserRequest(payLoad);
+    } catch (error) {
+      setUsers([...users]);
+    }
+  };
+
+  return (
+    <Fragment>
+      <Modal show={showModal} onHide={() => setShowModal(false)}>
+        <Modal.Header closeButton>
+          <Modal.Title>{title}</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <label className="text-start my-1">
+            Name:
+            <input
+              className="form-control"
+              value={user?.name}
+              onChange={(event) =>
+                setUser({ ...user, name: event.target.value })
+              }
+            ></input>
+          </label>
+          <label className="text-start my-1">
+            Username:
+            <input
+              className="form-control"
+              value={user?.username}
+              onChange={(event) =>
+                setUser({
```

Comment:
create a single onChange handler for all of the input fields

Reply:
do we still need it if it is used only once in the FormInput.js component -which we are using for each form input.